core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java

```diff
@@ -339,6 +339,177 @@ private void runScenarioTests(TestPlan testPlan, DeploymentCreationResult deploy
         }
     }
 
+    /**
+     * Execute given test scenario
+     *
+     * @param testScenario the test scenario
+     * @param deploymentCreationResult the result of the previous build step
+     * @param testPlan the test plan
+     */
+    private void executeTestScenario(TestScenario testScenario, DeploymentCreationResult deploymentCreationResult,
+                                     TestPlan testPlan) {
+        try {
+            scenarioExecutor.execute(testScenario, deploymentCreationResult, testPlan);
+            Optional<ResultParser> parser = ResultParserFactory.getParser(testPlan, testScenario);
+            if (parser.isPresent()) {
+                try {
+                    ResultParser resultParser = parser.get();
+                    resultParser.parseResults();
+                    resultParser.persistResults();
+                } catch (ResultParserException e) {
+                    logger.error(String.format("Error parsing the results for the" +
+                            " scenario %s", testScenario.getName()));
+                }
+            } else {
+                testScenario.setStatus(Status.ERROR);
+                logger.error(String.format("Error parsing the results for the " +
+                        "scenario %s", testScenario.getName()));
+            }
+        } catch (ScenarioExecutorException e) {
+            // we need to continue the rest of the tests irrespective of the exception thrown here.
+            logger.error(StringUtil.concatStrings("Error occurred while executing the SolutionPattern '",
+                    testScenario.getName(), "' in TestPlan\nCaused by "), e);
+        }
+        try {
+            persistTestScenario(testScenario);
+        } catch (TestPlanExecutorException e) {
+            logger.error(StringUtil.concatStrings(
+                    "Error occurred while persisting test scenario ", testScenario.getName()), e);
+        }
+    }
+
+    /**
+     * Apply config change set script before and after run test scenarios
+     *
+     * @param testPlan the test plan
+     * @param configChangeSet   config change set
+     * @param isInit run apply config-script if true. else, run revert-config script
+     * @return execution passed or failed
+     */
+    private boolean applyConfigChangeSet(TestPlan testPlan, ConfigChangeSet configChangeSet, boolean isInit) {
```

Comment:
Shall we hand-over the handling of configChangeSets to its own class? That'll improve the readability a lot imo.

Reply:
+1